apps/backend/src/modules/projects/project.repository.ts

```diff
@@ -75,6 +75,22 @@ class ProjectRepository implements Repository {
 		return item ? ProjectEntity.initialize(item) : null;
 	}
 
+	public async findProjectsByContributorId(
+		contributorId: number,
+	): Promise<{ items: ProjectEntity[] }> {
+		const projects = await this.projectModel
+			.query()
+			.select("projects.*")
+			.join("activity_logs", "activity_logs.project_id", "projects.id")
+			.join("git_emails", "git_emails.id", "activity_logs.git_email_id")
+			.where("git_emails.contributor_id", contributorId)
+			.distinct();
```

Comment:
so you won't need this scary query ;)

Reply:
It'll be handled by objection's `.withGraphFetched()` method. At least I think so